### DISCUSSION TITLE
Cypress setup for Mono Repo ?

### DISCUSSION THREAD

**Discussion by elamurugan-r** [cypress-io/cypress#22508] [Questions and Help]
### How to setup Cypress for a mono-repo?

- I have 3 React modules combined together gives a mono-epo using Yarn workspace.
- Each module has its own configuration like babel, webpack, tsconfig, package.json etc..,
- And the **mono-repo** has only package.json and tsconfig.json which refers all the modules' tsconfig,json files

The project structure:
![image](https://user-images.githubusercontent.com/86850779/175546683-561a24c9-d62a-494c-98a6-27bed530d7c8.png)

### Questions:

> **Do i need to setup cypress for each module (repo1, repo2, repo3) seperately ?**

                                                        or

> **Do i need to setup cypress for the common directory  MONO_REPO?**

**Comment by websitevirtuoso** [comment]
I think, it depends on your preferences.
If you repo1 or repo2. can leave independent (like modular arhjitecture then yes you should init it inside)  if not then use only 2th variant
I dont use monorepo but my modules looks like

![image](https://user-images.githubusercontent.com/5112918/175756584-8df8a7f0-8b0d-4919-b370-cc6879df328b.png)
